### PR TITLE
Add Type Declarations

### PR DESCRIPTION
**Pull request by philippfromme**
Requires https://github.com/bpmn-io/bpmn-js/pull/1899.

### CODE REVIEW

**Comment by nikku** [comment]
@philippfromme I believe JSON parsing is best handled like this: https://github.com/camunda/camunda-bpmn-js/pull/271/commits/9077ea856102089247d4871dd87e772e61afb38a.

File: package.json [hunk 1]
```diff
@@ -20,8 +21,11 @@
     "start:cloud-viewer": "cross-env SINGLE_START=camunda-cloud-viewer npm run dev",
     "start:platform-navigated-viewer": "cross-env SINGLE_START=camunda-platform-navigated-viewer npm run dev",
     "start:cloud-navigated-viewer": "cross-env SINGLE_START=camunda-cloud-navigated-viewer npm run dev",
-    "prepublishOnly": "run-s test:distro",
-    "prepare": "run-s clean distro"
+    "prepublishOnly": "run-s generate-types test:distro",
+    "prepare": "run-s clean distro",
+    "generate-types": "run-s generate-types:*",
+    "generate-types:generate": "del-cli \"lib/**/*.d.ts\" && npx bio-dts -r --resolveJsonModule --esModuleInterop lib",
```
Comment (nikku) [inline package.json:27]: Do we want to depend on an explicit version of `bio-dts`, i.e. through installation:

```sh
npm install bio-dts -D
```

Comment (philippfromme) [reply package.json:27]: If we decide to, we should make it explicit in diagram-js and bpmn-js, too.

**Review by nikku — APPROVED** [review]
Great stuff :heart_eyes:

**Pull request merged**